models/path.py

```diff
@@ -147,6 +147,8 @@ def get_paths(circuit, max_paths=2, **kwargs) -> list[dict]:
             api_reply = httpx.post(endpoint, json=request_data, timeout=10)
         except httpx.TimeoutException as err:
             raise PathFinderException(str(err)) from err
+        except httpx.ConnectError as err:
```

Comment:
This is a bit of improvement. But, notice that it still vulnerable to other types of Connection errors like `[Read|Write|Close]Error`. It's usually the way to go to handle `httpx.RequestError` that covers many of these underlying transitive exceptions, while most of them are retryable, and even for those that there aren't it won't cause further harm if they get retried, so this should provide stronger reliablity retries for us. See what I mean? Also, since you're at it, and this is very safe and small change let's ship as a patch too. 

Here's httpx exc hierarchy: https://github.com/encode/httpx/blob/master/httpx/_exceptions.py#L5

Reply:
Changed. Keeping the test unchanged.